src/server/standalone.js

```diff
@@ -401,19 +407,33 @@ class StandAloneServer {
                         }
                     });
             } else if (req.cookies[__gmeConfig.authentication.jwt.cookieId]) {
+                //TODO this is a forceful method of trying to renew aad token needs to be harmonized better
+                let webgmeTokenResult = null;
                 __logger.debug('jwtoken provided in cookie');
                 token = req.cookies[__gmeConfig.authentication.jwt.cookieId];
+                const aToken = req.cookies[__gmeConfig.authentication.azureActiveDirectory.cookieId];
                 __gmeAuth.verifyJWToken(token)
                     .then(result => {
-                        if (result.renew === true) {
+                        webgmeTokenResult = result;
+                        if (__gmeConfig.authentication.azureActiveDirectory.enable) {
+                            return this.__aadClient.getAccessToken(result.content.userId, aToken);   
```

Comment:
This would require the clients to have both a webgme-token and a azure-token. Is that really necessary?

Reply:
I realize that it is - since this implementation is not only for user authentication, but can also grant access (for plugins) to azure services.  
The second part is great to have, but would it possible to split these up? Not code-wise but configuration wise? If you only want the additional authentication then it would make sense to only ensure there is a token. I haven't tried this out, but maybe the configuration on azure is more straight forward.

Reply:
I see that `accessScope` controls this and is null by default.

Reply:
if we say they are not 'required' to have aad token, how and when are we actually going to get/use it?
my issue of allowing not having an aad token is the unpredictability that the user of said token might experience...
now we can control this behavior by saying that we only require 'access' tokens to be there all the time, but then the question is still remains, who and when will actually authenticate with azure after the initial login?